common/src/main/java/org/apache/sedona/common/utils/RasterUtils.java

```diff
@@ -347,4 +348,19 @@ public static int getDataTypeCode(String s) {
         }
         return 5; // defaulting to double
     }
+
+    public static boolean isDataTypeIntegral(int dataTypeCode) {
+        //returns true if the datatype code refers to an int-like datatype (int, short, etc)
+        switch (dataTypeCode) {
+            case 3: //int
```

Comment:
I suspect this logic will fail when you give short or byte. Can you try to add test cases for shorts/byte?